src/Liip/RMT/Config/Handler.php

```diff
@@ -129,6 +129,18 @@ protected function validateRootElements($config)
                 throw new Exception("[$mandatoryParam] should be defined");
             }
         }
+
+        if ($config['version-generator'] == 'simple'
+            && isset($config['vcs']) && $config['vcs'] == 'hg'
+            && (! isset($config['version-persister']['tag-prefix']) || strlen(trim($config['version-persister']['tag-prefix'])) == 0))
+        {
+            exec('hg version', $result);
+            preg_match('/version (\d+)\.\d+\.\d+/', reset($result), $matches);
+            if (intval($matches[1]) >= 3) {
+                throw new Exception("Simple generator is not compatible with mercurial > 3.");
+            }
+
+        }
```

Comment:
I don't know if it's the right place to put this mercurial specific code... At first I was more expecting a new method on the VCS interface. Something like `supportTaggingWithInteger()`. But then the error have to be raised after the config have been parsed, at the beginning of the `release`action...